core/src/main/java/jenkins/model/LoginPageDecorator.java

```diff
@@ -0,0 +1,32 @@
+package jenkins.model;
+
+import hudson.DescriptorExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+
+public class LoginPageDecorator extends Descriptor<LoginPageDecorator> implements ExtensionPoint, Describable<LoginPageDecorator> {
+
+    protected LoginPageDecorator()  {
+        super(self());
+    }
+
+    @Override
+    public final Descriptor<LoginPageDecorator> getDescriptor() {
+        return this;
+    }
+
+    public final String getUrl() {
+        return "descriptor/"+clazz.getName();
+    }
+
+    public static LoginPageDecorator first(){
+        if(Jenkins.getInstanceOrNull()==null) {
```

Comment:
I think this check is not needed, but if it is then mark the method as `@CheckForNull`